src/common/transformations/src/transformations/common_optimizations/nop_elimination.cpp

```diff
@@ -477,6 +478,146 @@ pass::EliminateSqueeze::EliminateSqueeze() {
+            if (axis_val.size() == 1 && axis_val[0] == 1) {
+                return true;
+            }
+        }
+    }
+    return false;
+}
+
+bool check_reshape(const shared_ptr<Node>& node) {
+    auto reshape = std::dynamic_pointer_cast<ov::opset9::Reshape>(node);
+    if (reshape) {
+        auto shape_pattern =
+            std::dynamic_pointer_cast<ov::opset9::Constant>(reshape->input_value(1).get_node_shared_ptr());
+        if (shape_pattern) {
+            auto pattern_val = shape_pattern->cast_vector<int64_t>();
+            pattern_val.insert(pattern_val.begin() + 1, 1);
+            auto in_shape = reshape->input_value(0).get_partial_shape();
+            if (in_shape.compatible(pattern_val)) {
+                return true;
+            }
+        }
+    }
+    return false;
+}
+
+template <class T>
+std::shared_ptr<T> check_all_inputs(const std::shared_ptr<ov::opset9::Concat>& concat) {
+    shared_ptr<T> split;
+    const auto concat_in_values = concat->input_values();
+    int idx = 0;
+    for (const auto& in_to_concat : concat_in_values) {
+        const auto& cast_to_split = std::dynamic_pointer_cast<T>(in_to_concat.get_node_shared_ptr());
+        // There is a special case with (GRU/RNN/LSTM)Sequence ops
+        // Sequence->output(1) can be connected directly to the last input to Concat
+        // and the last Split output is not used. This is also a valid case for this Elimination.
+        if (!cast_to_split) {
+            if (idx != (concat_in_values.size() - 1) || !split) {
+                return {};
+            }
+            shared_ptr<Node> in_to_split = split->input_value(0).get_node_shared_ptr();
+            Output<Node> seq_out;
+            if (!in_to_split->inputs().empty()) {
+                seq_out = in_to_split->input_value(0);
+            } else {
+                return {};
+            }
+
+            auto seq_node = seq_out.get_node_shared_ptr();
+            if (seq_out.get_index() != 0 || !(dynamic_pointer_cast<ov::opset9::RNNSequence>(seq_node) ||
+                                              dynamic_pointer_cast<ov::opset9::GRUSequence>(seq_node) ||
+                                              dynamic_pointer_cast<ov::opset9::LSTMSequence>(seq_node))) {
+                return {};
+            }
+
+            // check that Split is connected to Sequence->output(0)
+            // possible patterns:
+            // Sequence:0->Squeeze->Split
+            bool valid_pattern = check_squeeze(in_to_split);
+            // Sequence:0->Reshape->Split
+            if (!valid_pattern) {
+                valid_pattern = check_reshape(in_to_split);
+            }
+
+            if (!valid_pattern) {
+                return {};
+            }
+
+            // check that Sequence->output(1) is connected to this input
+            if (in_to_concat != seq_node->output(1)) {
+                return {};
+            }
+            return split;
```

Comment:
Shouldn't we check other inputs to concat?

Reply:
we shouldn't. This check is necessary for this special case only, we need to check that this in_to_concat is exactly seq_node->output(1). 
All other inputs to Concat op should be connected to Split (VariadicSplit) and we already have these checks.